ecl/eclccserver/eclccserver.cpp

```diff
@@ -881,7 +918,22 @@ class EclccServer : public CInterface, implements IThreadFactory, implements IAb
     bool running;
     CSDSServerStatus serverstatus;
     Owned<IJobQueue> queue;
+    StringAttr updatedQueueNames;
+    unsigned reloadConifgCBId = 0;
 
+
+    void configUpdate()
+    {
+        StringBuffer newQueueNames;
+        getQueues(newQueueNames);
+        if (!newQueueNames.length())
+            ERRLOG("No queues found to listen on");
+        if (!strsame(queueNames, newQueueNames))
+        {
+            updatedQueueNames.set(newQueueNames);
```

Comment:
Does this need protecting from a race with the thread that reads it?

Reply:
theoretically it could race with another call to update, have protected and rearrange slightly (so more common code)

Reply:
It still needs a cs - since this thread could be updating updatedQueueNames at the same time as it is being checked.

Reply:
whoops - I was supposed to add there, as part of this change.